### PR TITLE
Move away from object templates, part 1

### PR DESCRIPTION
**Pull request by hermanventer**
This is the first in many steps needed to get rid of object templates for abstract objects.

When we get there, AbstractObjectValue will represent either a set of concrete objects, or a completely unknown object. Since any operation performed on the latter will have unknown effects on the heap, most operations will become errors. I.e. completely unknown objects can be assigned to other properties and tested for membership of Object and pretty much nothing else.

Most of the current uses of AbstractObjectValue + template will have to become uses of concrete objects that have been marked as partial and/or simple.

Currently, not all such uses can be simply transformed in this way, because there are  lot of places where checks depend on an object being abstract. This is actually a pretty bad bug, since partial concrete objects are just as problematic. This will be addressed in a series of subsequent pull requests.

The relevant is issue is #882.

### CODE REVIEW

File: src/intrinsics/node/utils.js [hunk 1]
```diff
@@ -24,7 +24,7 @@ export function getNodeBufferFromTypedArray(realm: Realm, value: ObjectValue): U
 // Takes a value from the host realm and create it into a Prepack Realm.
 // TODO: Move this to a bigger general purpose proxy between the environments.
 // See issue #644 for more details.
-export function createDeepIntrinsic(realm: Realm, value: Value, intrinsicName: string) {
+export function createDeepIntrinsic(realm: Realm, value: any, intrinsicName: string) {
```
Comment (NTillmann) [inline src/intrinsics/node/utils.js]: Why this change? Looks like this might be related to a new Flow version (https://github.com/facebook/prepack/issues/905). If so, I'd prefer a separate pull request that bumps the Flow version number as wel.

File: src/intrinsics/prepack/global.js [hunk 2]
```diff
@@ -221,11 +221,11 @@ export default function(realm: Realm): void {
 
   // __makePartial(object) marks an (abstract) object as partial.
   global.$DefineOwnProperty("__makePartial", {
-    value: new NativeFunctionValue(realm, "global.__makePartial", "__isPartial", 1, (context, [object]) => {
+    value: new NativeFunctionValue(realm, "global.__makePartial", "__makePartial", 1, (context, [object]) => {
```
Comment (NTillmann) [inline src/intrinsics/prepack/global.js]: Good catch.

File: src/intrinsics/node/utils.js (hunk 1 above)
Comment (hermanventer) [inline src/intrinsics/node/utils.js]: Actually, it should be "mixed" not "any". I'll remove this.
